src/commands/lint.ts

```diff
@@ -17,117 +17,182 @@
+   *   - --watch
+   *   - --fix
+   */
+
+  async run(options: Options, config: ProjectConfig): Promise<any> {
+    // Defer dependency loading until this specific command is run.
+    const lintLib: typeof lintLibTypeOnly = require('polymer-linter');
+    const Analyzer: typeof AnalyzerTypeOnly =
+        require('polymer-analyzer').Analyzer;
+    const FSUrlLoader: typeof FSUrlLoaderTypeOnly =
+        require('polymer-analyzer/lib/url-loader/fs-url-loader').FSUrlLoader;
+    const PackageUrlResolver: typeof PackageUrlResolverTypeOnly =
+        require('polymer-analyzer/lib/url-loader/package-url-resolver')
+            .PackageUrlResolver;
+    const Severity: typeof SeverityTypeOnly =
+        require('polymer-analyzer/lib/warning/warning').Severity;
+    const WarningFilter: typeof WarningFilterTypeOnly =
+        require('polymer-analyzer/lib/warning/warning-filter').WarningFilter;
+    const WarningPrinter: typeof WarningPrinterTypeOnly =
+        require('polymer-analyzer/lib/warning/warning-printer').WarningPrinter;
 
-    if (lintFiles.length === 0) {
+    this._loadPlugins(config);
+
+    const lintOptions: Partial<typeof config.lint> = (config.lint || {});
+
+    const ruleCodes = options.rules || lintOptions.rules;
+    if (ruleCodes === undefined) {
       logger.warn(
-          'No inputs specified. Please use the --input, --entrypoint, ' +
-          '--shell or --fragment flags');
-      let argsCli = commandLineArgs(this.args);
-      console.info(argsCli.getUsage({
-        title: `polymer ${this.name}`,
-        description: this.description,
-      }));
-      return Promise.resolve();
+          `You must state which lint rules to use. You can use --rules, ` +
+          `but for a project it's best to use polymer.json. e.g.
+
+{
+  "lint": {
+    "rules": ["polymer-2"]
+  }
+}`);
+      process.exitCode = 1;
+      return;
     }
 
-    // Default to false if input files are provided, otherwise default to true
-    let followDependencies = !options['input'];
-    if (options['follow-dependencies']) {
-      followDependencies = true;
-    } else if (options['no-follow-dependencies']) {
-      followDependencies = false;
+    const rules = lintLib.registry.getRules(ruleCodes || lintOptions.rules);
+    const filter = new WarningFilter({
+      warningCodesToIgnore: new Set(lintOptions.ignoreWarnings || []),
+      minimumSeverity: Severity.WARNING
+    });
+
+
+    const analyzer = new Analyzer({
+      urlLoader: new FSUrlLoader(config.root),
+      urlResolver: new PackageUrlResolver(),
+    });
+
+    const linter = new lintLib.Linter(rules, analyzer);
+
+    const warnings = await this._lint(linter, options.input);
+
+    const filtered = warnings.filter((w) => !filter.shouldIgnore(w));
+
+    const printer = new WarningPrinter(
+        process.stdout, {analyzer: analyzer, verbosity: 'full', color: true});
+    await printer.printWarnings(filtered);
+
+    process.exitCode = this._getExitCode(filtered);
```

Comment:
There is already error handling in place to log the error and return the right exit code on `run()` rejection. I'd much prefer that we use that flow here as well instead of modifying the `process` object directly.

When multiple build errors happen, an error with a message like "3 error(s) occurred during build" is thrown and logged. Something similar with `filtered.length` could work well here.